libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificatePinnerFactory.java

```diff
@@ -36,11 +38,39 @@ CertificatePinner provideCertificatePinnerFor(Environment environment) {
     return CERTIFICATES_PINS.get(environment);
   }
 
-  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder) {
+  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder,
+                                   Environment environment, CertificateBlacklist certificateBlacklist) {
+    pins = removeBlacklistedPins(pins, environment, certificateBlacklist);
+
     for (Map.Entry<String, List<String>> entry : pins.entrySet()) {
       for (String pin : entry.getValue()) {
         builder.add(entry.getKey(), pin);
       }
     }
   }
+
+  private Map<String, List<String>> removeBlacklistedPins(Map<String, List<String>> pins, Environment environment,
+                                                          CertificateBlacklist certificateBlacklist) {
+    String key = COM_EVENTS;
+
+    if (environment == Environment.CHINA) {
+      key = CHINA_EVENTS;
+    }
+
+    List<String> hashList = pins.get(key);
```

Comment:
At this point we could obtain the key from the `pins` as follows:

```java
Set<String> pinsKey = pins.keySet();
String key = pinsKey.iterator().next();
```

We can guarantee that the `pins` instance has only one entry so we don't need to check for the key explicitly - `Environment` could be removed all the way up and `COM_EVENTS` and  `CHINA_EVENTS` constants too.